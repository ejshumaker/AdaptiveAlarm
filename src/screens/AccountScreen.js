/**
 * BASED OFF OF HOMESCREEN by:
 * @eschirtz 03-03-19
 *
 * Edited / Designed by:
 * @weinoh 03-28-19
 */
import React, { Component } from 'react';
import {
  View, Text, Image, Button, StyleSheet, ActivityIndicator,
} from 'react-native';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { userSignOut } from '../store/actions/userActions';
import { alarmCalculateTime } from '../store/actions/alarmActions';

import {
  Colors,
  GlobalStyles,
} from '../constants';

// STYLESHEET FOR USER PROFILE
const styles = StyleSheet.create({
  titleView: {
    marginTop: 40,
    marginBottom: 10,
  },
  imageView: {
    width: '80%',
    margin: 10,
    alignItems: 'center',
    justifyContent: 'center',
  },
  usericon: {
    width: 150,
    height: 150,
    borderRadius: 75,
    margin: 20,
  },
  userinfopane: {
    width: '95%',
    alignItems: 'center',
    justifyContent: 'space-around',
    flexDirection: 'column',
  },
  profileRow: {
    flex: 8,
    flexDirection: 'row',
    justifyContent: 'space-between',
    borderBottomWidth: 2,
    borderBottomColor: Colors.white,
    marginBottom: 10,
    height: 32,
    width: '90%',
  },
  infoColumn: {
    flexDirection: 'row',
    justifyContent: 'flex-start',
  },
  dataColumn: {
    flexDirection: 'row',
    justifyContent: 'flex-end',
  },
  signOutButton: {
    width: '80%',
    justifyContent: 'space-around',
    height: 80,
    marginBottom: 50,
  },
});

class AccountScreen extends Component {
  constructor() {
    super();
    this.state = {
      title: 'YOUR PROFILE:',
    };
  }

  loader() {
    const { loading } = this.props;
    if (loading) {
      return <ActivityIndicator color={Colors.primary} size="large" />;
    } return null;
  }

  render() {
    const { title } = this.state;
    const {
      // navigation, // from react-navigation
      signOut, // Redux actions
      // calculateTime,
      firstName, // Redux store
      lastName,
      // userName,
      // email,
      // alarmTime,
      // errorMessage,
    } = this.props;
    // const { navigate } = navigation;

    return (
      <View style={[GlobalStyles.centerChildrenXY, { height: '85%', justifyContent: 'space-around' }]}>


        {/* VIEW FOR TITLE ! "YOUR PROFILE" */}
        <View style={[styles.titleView, { flex: 1 }]}>
          <Text style={[
            GlobalStyles.h2,
            { color: Colors.primary },
          ]}
          >
            {title}
          </Text>


          {/* VIEW FOR IMAGE OF USER */}
          <View style={[styles.imageView, { flex: 1 }]}>


            {/* TODO: Replace with Icon ! */}
            <Image
              source={{ uri: 'https://reactnativecode.com/wp-content/uploads/2018/01/2_img.png' }}
              style={styles.usericon}
            />
            { /* LOADING ICON FUNCTION */ }
            { this.loader() }

            { /* DISPLAY USER'S NAME BELOW IMAGE */ }
            <Text style={[GlobalStyles.h3, { color: Colors.primary }]}>
              {firstName}
              {' '}
              {lastName}
            </Text>
          </View>
          {/* END -- VIEW FOR IMAGE OF USER */}

        </View>
        { /* END -- VIEW FOR TITLE */ }

        {/* VIEW FOR USER INFO */}
        <View style={[styles.userinfopane, { flex: 1 }]}>


          {/* VIEW FOR USER'S ROUTINE TIME */}
          <View style={[styles.profileRow, { flex: 1 }]}>

            <View style={styles.infoColumn}>
              <Text style={[GlobalStyles.paragraph]}>Routine Time</Text>
            </View>

            <View style={styles.dataColumn}>
              <Text style={[
                GlobalStyles.paragraph,
                {
                  color: Colors.primary,
                },
              ]}
              >
                {'45 minutes'}
              </Text>
            </View>
          </View>
          {/* END --  VIEW FOR USER'S ROUTINE TIME */}

          {/* VIEW FOR USER'S HOME ADDRESS */}
          <View style={[styles.profileRow, { flex: 1 }]}>

            <View style={styles.infoColumn}>
              <Text style={[GlobalStyles.paragraph]}>Home</Text>
            </View>

            <View style={styles.dataColumn}>
              <Text style={[
                GlobalStyles.paragraph,
                {
                  color: Colors.primary,
                },
              ]}
              >
                {/* TODO: REPLACE WITH CURRENT LOCATION (ADDRESS) */}
                {'210 Lakelawn Place, Madison, WI'}
              </Text>
            </View>

          </View>
          {/* END --  VIEW FOR USER'S HOME ADDRESS */}


          {/* VIEW FOR USER'S WORK ADDRESS */}
          <View style={[styles.profileRow, { flex: 1 }]}>

            <View style={styles.infoColumn}>
              <Text style={[GlobalStyles.paragraph]}>Work</Text>
            </View>

            <View style={styles.dataColumn}>
              <Text style={[
                GlobalStyles.paragraph,
                {
                  color: Colors.primary,
                },
              ]}
              >
                {/* TODO: REPLACE WITH ACTUAL DESTINATION INPUT */}
                {'1308 W Dayton St, Madison, WI'}
              </Text>
            </View>
          </View>
          {/* END -- VIEW FOR USER'S WORK ADDRESS */}


        </View>
        {/* END -- VIEW FOR USER'S INFO */}


        {/* VIEW FOR SIGN OUT BUTTON */}
        <View style={styles.signOutButton}>
          <Button
            title="Sign Out"
            color={Colors.darkGray}
            onPress={signOut}
          />
        </View>
        {/* END -- VIEW FOR SIGN OUT BUTTON */}


      </View>
    );
    // END RENDER, ABOVE IS CLOSING VIEW TAG.
  }
}

AccountScreen.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
  }).isRequired,
  // Redux state
  firstName: PropTypes.string,
  lastName: PropTypes.string,
  // userName: PropTypes.string,
  // email: PropTypes.string,
  // errorMessage: PropTypes.string,
  loading: PropTypes.bool.isRequired,
  // alarmTime: PropTypes.number.isRequired,
  // // Redux dispatch
  // calculateTime: PropTypes.func.isRequired,
  signOut: PropTypes.func.isRequired,
};

AccountScreen.defaultProps = {
  firstName: '',
  lastName: '',
  // userName: '',
  // email: '',
  // errorMessage: '',
};

/**
 * Pull in only the fields you need from
 * the store. They are then accesible via 'props'
 * @eschirtz 03-03-19
 */
const mapStateToProps = state => ({
  firstName: state.user.firstName,
  lastName: state.user.lastName,
  userName: state.user.userName,
  email: state.user.email,
  errorMessage: state.user.errorMessage,
  loading: state.user.loadingFetch,
  alarmTime: state.alarm.time,
});

/**
 * Assign the action creators to props,
 * import actions at the top of the file
 * @eschirtz 03-03-19
 */
const mapDispatchToProps = dispatch => ({
  signOut: () => { dispatch(userSignOut()); },
  calculateTime: (time) => { dispatch(alarmCalculateTime(time)); },
});

export default connect(mapStateToProps, mapDispatchToProps)(AccountScreen);
export { AccountScreen };
