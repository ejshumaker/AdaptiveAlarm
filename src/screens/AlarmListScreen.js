import React, { Component } from 'react';
import { connect } from 'react-redux';
import {
  View, Text, FlatList, ActivityIndicator, TouchableOpacity,
} from 'react-native';
import PropTypes from 'prop-types';

import { GlobalStyles, Colors } from '../constants';

import { AlarmItem, StatusBarBackground } from '../components';
import { CloseIcon } from '../icons/close';

import { userSetAlarmStatus } from '../store/actions/userActions';


class AlarmListScreen extends Component {
  constructor() {
    super();
    this.state = {
    };
  }

  // Loading icon !
  loader() {
    const { loading } = this.props;
    if (loading) {
      return <ActivityIndicator color={Colors.primary} size="large" />;
    }
    return null;
  }

  menu() {
    const {
      navigation,
    } = this.props;
    const { navigate } = navigation;
    return (
      <View style={GlobalStyles.menu}>
        <TouchableOpacity
          onPress={() => { navigate('Main'); }}
        >
          <CloseIcon />
        </TouchableOpacity>
      </View>
    );
  }


  render() {
    const { alarms, toggleAlarm, navigation } = this.props;
    const { navigate } = navigation;

    return (
      <View style={{ flex: 1 }}>
        <StatusBarBackground />
        {this.menu()}
        <View style={[GlobalStyles.container, { paddingHorizontal: 48 }]}>
          <Text
            style={[
              GlobalStyles.h2,
              {
                color: Colors.primary,
                marginBottom: 48,
              },
            ]}
          >
          ALL ALARMS:
          </Text>
          {this.loader()}

          <FlatList
            showsHorizontalScrollIndicator={false}
            showsVerticalScrollIndicator={false}
            data={Object.values(alarms)}
            renderItem={({ item, index }) => (
              <AlarmItem
                alarm={item}
                alarmId={Object.keys(alarms)[index]}
                toggleAlarm={toggleAlarm}
                navigate={navigate}
              />
            )}
            keyExtractor={(item, index) => String(index)}
          />

        </View>
      </View>
    );
  }
  // end render
}
// end class

// I think this is almost completed ?
AlarmListScreen.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
  }).isRequired,
  // Redux state
  loading: PropTypes.bool.isRequired,
  alarms: PropTypes.object, // eslint-disable-line
  // Redux dispatch
  toggleAlarm: PropTypes.func.isRequired,
};

AlarmListScreen.defaultProps = {
  alarms: {},
};

/**
 * @weinoh 04-10-19
 */
const mapStateToProps = state => ({
  alarmTime: state.alarm.time,
  alarmCalculating: state.alarm.loading,
  loading: state.user.loadingFetch,
  alarms: state.user.alarms,
});

/**
 * TODO: Does this toggleAlarm function need to be passed into our alarmitem?
 * @weinoh 04-10-19
 */
const mapDispatchToProps = dispatch => ({
  toggleAlarm: (alarmid, status) => {
    dispatch(userSetAlarmStatus(alarmid, status));
  },
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(AlarmListScreen);
