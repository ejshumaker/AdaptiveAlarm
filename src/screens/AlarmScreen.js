import React, { Component } from 'react';
import { connect } from 'react-redux';
import {
  View, Text, TouchableOpacity,
} from 'react-native';
import moment from 'moment';
import PropTypes from 'prop-types';
import store from '../store';
import { alarmCalculateTime } from '../store/actions/alarmActions';

import { GlobalStyles, Colors } from '../constants';
import { Buttons, StatusBarBackground } from '../components';
import { CloseIcon } from '../icons/close';

import Alarm from '../custom_modules/Alarm';

class AlarmScreen extends Component {
  constructor() {
    super();
    this.state = {
      time: moment().format('LT'),
    };
  }

  componentDidMount() {
    this.setState({
      time: moment().format('LT'),
    });
  }

  stopSound = (navigate) => {
    Alarm.stopAlarm();
    navigate('Main');
    console.log('should have navigated');
    store.dispatch(alarmCalculateTime());
  }

  menu() {
    const {
      navigation,
    } = this.props;
    const { navigate } = navigation;
    return (
      <View style={[GlobalStyles.menu]}>
        <TouchableOpacity
          onPress={() => { this.stopSound(navigate); }}
        >
          <CloseIcon />
        </TouchableOpacity>
      </View>
    );
  }

  render() {
    const { time } = this.state;
    const { navigation } = this.props;
    const { navigate } = navigation;
    return (
      <View>
        <StatusBarBackground />
        {this.menu()}
        <View style={{ alignItems: 'center', marginTop: 200 }}>
          <Text style={[GlobalStyles.h1, GlobalStyles.margin, { color: Colors.primary }]}>
            {time}
          </Text>
          <View style={{ height: 8, width: 8 }} />
          <Buttons
            title="STOP"
            backgroundColor={Colors.darkGray}
            textColor={Colors.white}
            onPress={() => this.stopSound(navigate)}
          />
        </View>
      </View>
    );
  }
}

AlarmScreen.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
  }).isRequired,
  alarm: PropTypes.object.isRequired, // eslint-disable-line
};

const mapStateToProps = state => ({
  alarm: state.alarm,
});

AlarmScreen.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
    addListener: PropTypes.func.isRequired,
  }).isRequired,
};

export { AlarmScreen };
export default connect(mapStateToProps)(AlarmScreen);
