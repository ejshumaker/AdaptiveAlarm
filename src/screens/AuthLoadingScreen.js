import React from 'react';
import {
  ActivityIndicator,
  View,
} from 'react-native';
import { connect } from 'react-redux';
import { auth } from 'firebase';
import PropTypes from 'prop-types';

import { userFetch } from '../store/actions/userActions';
import { GlobalStyles, Colors } from '../constants';

class AuthLoadingScreen extends React.Component {
  constructor(props) {
    super(props);
    this.bootstrapAsync();
  }

  // Fetch the token from storage then navigate to our appropriate place
  bootstrapAsync = async () => {
    const { navigation, fetchUser } = this.props;

    // This will switch to the App screen or Auth screen and this loading
    // screen will be unmounted and thrown away.
    auth().onAuthStateChanged((user) => {
      if (user) {
        // User is signed in.
        fetchUser(user.uid, navigation.navigate);
        navigation.navigate('App');
      } else {
        // User is signed out.
        navigation.navigate('Auth');
      }
    });
  };

  // Render simple spinner,
  // this is usually so fast you won't see it
  render() {
    return (
      <View style={GlobalStyles.centerChildrenXY}>
        <ActivityIndicator color={Colors.primary} size="large" />
      </View>
    );
  }
}

AuthLoadingScreen.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
  }).isRequired,
  // Redux dispatch
  fetchUser: PropTypes.func.isRequired,
};

const mapStateToProps = () => ({});

const mapDispatchToProps = dispatch => ({
  fetchUser: (user, navigate) => { dispatch(userFetch(user, navigate)); },
});

export { AuthLoadingScreen };
export default connect(mapStateToProps, mapDispatchToProps)(AuthLoadingScreen);
