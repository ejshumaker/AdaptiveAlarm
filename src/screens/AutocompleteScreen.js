/**
  * Uses Google Maps API to autocomplete search entries.
  * Code modified from Stefan Hyltoft (https://github.com/Hyllesen)
  * @weinoh 03-26-2019
  * @ejshumaker 03-24-2019
*/

import React, { Component } from 'react';
import { View } from 'react-native';
import { GlobalStyles } from '../constants';

import Autocomplete from '../components/Autocomplete';

export default class AutocompleteScreen extends Component {

  render() {
    return (
      <View style={GlobalStyles.centerChildrenXY}>
        <Autocomplete />
      </View>
    );
  }
}
