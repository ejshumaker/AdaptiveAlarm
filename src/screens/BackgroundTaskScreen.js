/**
  * Test screen to fire off background task
  * @ejshumaker 03-24-2019
*/

import React, { Component } from 'react';
import {
  View,
  Text,
  AppState,
  AsyncStorage,
} from 'react-native';
import { NavigationEvents } from 'react-navigation';
import { TaskManager, BackgroundFetch, Location } from 'expo';
import { GlobalStyles, Colors } from '../constants';
import Buttons from '../components/Buttons';

const BACKGROUND_FETCH_TASK = 'background-fetch';

// eslint-disable-next-line
export default class BackgroundTaskScreen extends Component {
  state = {
    status: null,
    isRegistered: false,
  };

  componentDidMount() {
    AppState.addEventListener('change', this.handleAppStateChange);
  }

  componentWillUnmount() {
    AppState.removeEventListener('change', this.handleAppStateChange);
  }

  didFocus = () => {
    this.checkStatusAsync();
  };

  handleAppStateChange = (nextAppState) => {
    if (nextAppState === 'active') {
      this.checkStatusAsync();
    }
  };

  toggle = async () => {
    await BackgroundFetch.setMinimumIntervalAsync(10);
    console.log('setMinimumIntervalAsync');
    if (this.state.isRegistered) {
      await BackgroundFetch.unregisterTaskAsync(BACKGROUND_FETCH_TASK);
    } else {
      await BackgroundFetch.registerTaskAsync(BACKGROUND_FETCH_TASK);
    }
    this.setState({ isRegistered: !this.state.isRegistered });
  };

  async checkStatusAsync() {
    const status = await BackgroundFetch.getStatusAsync();
    const isRegistered = await TaskManager.isTaskRegisteredAsync(BACKGROUND_FETCH_TASK);
    console.log('status: ', BackgroundFetch.Status[status]);
    console.log('isRegistered: ', isRegistered);
    this.setState({ status, isRegistered });
  }

  render() {
    return (
      <View style={GlobalStyles.centerChildrenXY}>
        <NavigationEvents onDidFocus={this.didFocus} />
        <View style={GlobalStyles.container}>
          <Text style={{ color: Colors.primary }}>
            Background fetch status:{' '}
            <Text style={{ color: 'red' }}>
              {this.state.status ? BackgroundFetch.Status[this.state.status] : null}
            </Text>
          </Text>
        </View>
        <Buttons
          title={
            this.state.isRegistered
              ? 'Unregister BackgroundFetch task'
              : 'Register BackgroundFetch task'
          }
          backgroundColor={Colors.darkGray}
          textColor={Colors.primary}
          onPress={this.toggle}
        />
      </View>
    );
  }
}

TaskManager.defineTask(BACKGROUND_FETCH_TASK, async () => {
  console.log('in BACKGROUND_FETCH_TASK');
  return BackgroundFetch.Result.NewData;
});
