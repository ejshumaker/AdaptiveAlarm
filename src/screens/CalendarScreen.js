/**
  * Test screen for Expo's Calendar API
  * @ejshumaker 04-07-2019
*/

import React, { Component } from 'react';
import { View, Text } from 'react-native';
import { Calendar, Permissions } from 'expo';
import { GlobalStyles, Colors } from '../constants';
import { Buttons } from '../components';

// eslint-disable-next-line
export default class CalendarScreen extends Component {

  constructor() {
    super();
    this.state = {
      eventStart: '',
    };
  }

  async getStartTime() {
    let timeUTC = 0;
    await Permissions.askAsync('calendar');
    const cals = await Calendar.getCalendarsAsync();
    // get all device calendar ids
    const data = cals.filter(item => item).map(({ id }) => ({ id }));
    const tomorrowMorning = new Date();
    const timeZoneOffset = (tomorrowMorning.getTimezoneOffset() / 60);
    tomorrowMorning.setDate(tomorrowMorning.getDate() + 1);
    tomorrowMorning.setHours((0 - timeZoneOffset), 0, 0, 0);
    const tomorrowNight = new Date();
    tomorrowNight.setDate(tomorrowNight.getDate() + 1);
    tomorrowNight.setHours((23 - timeZoneOffset), 59, 59, 0);
    // check all events for the following day and return earliest event start time
    Calendar.getEventsAsync(data, tomorrowMorning, tomorrowNight).then((response) => {
      // parse String to get in UTC format
      const startTime = response[0].startDate;
      const date = String(startTime).split('.');
      timeUTC = Date.parse(date[0]);
      this.setState({ eventStart: timeUTC });
    });
  }

  render() {
    const {
      eventStart,
    } = this.state;
    return (
      <View style={[GlobalStyles.centerChildrenXY, { margin: 20 }]}>
        <Buttons
          title="Get Next Start Time"
          backgroundColor={Colors.darkGray}
          textColor={Colors.white}
          onPress={() => this.getStartTime()}
        />
        <Text style={GlobalStyles.input}>
          {/* eslint-disable-next-line */}
          Next Event: { eventStart }
        </Text>
      </View>
    );
  }
}
