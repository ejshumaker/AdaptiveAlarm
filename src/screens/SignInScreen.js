import React from "react";
import {
  TextInput,
  Text,
  View,
  ActivityIndicator,
  StatusBar
} from "react-native";
import { connect } from "react-redux";
import PropTypes from "prop-types";

import { userSignIn } from "../store/actions/userActions";
import { Colors, GlobalStyles } from "../constants";
import Buttons from "../components/Buttons";
import { EmailIcon } from "../icons/email";
import { KeyIcon } from "../icons/key";

class SignInScreen extends React.Component {
  state = {
    email: "",
    password: ""
  };

  onChangeText(key, value) {
    this.setState({
      [key]: value
    });
  }

  loader() {
    const { loading } = this.props;
    if (loading) {
      return <ActivityIndicator color={Colors.primary} size="large" />;
    }
    return null;
  }

  render() {
    const { signIn, navigation, errorMessage } = this.props;
    const { email, password } = this.state;
    return (
      <View style={[GlobalStyles.container, { justifyContent: 'center', padding: 48 }]}>
        <StatusBar barStyle="light-content" />
        <Text
          style={[
            GlobalStyles.h2,
            { textAlign: "left", color: Colors.primary }
          ]}
        >
          SIGN IN:
        </Text>
        {this.loader()}
        <Text
          style={[
            GlobalStyles.paragraph,
            { color: Colors.error, marginVertical: 24 }
          ]}
        >
          {errorMessage}
        </Text>
        <View
          style={{
            flexDirection: "row",
            height: 40,
            backgroundColor: Colors.darkGray,
            borderRadius: 8,
            marginBottom: 30
          }}
        >
          <EmailIcon style={{ marginLeft: 13, marginTop: 7 }} />
          <TextInput
            onChangeText={value => this.onChangeText("email", value)}
            style={GlobalStyles.destinationInput}
            placeholder="Email"
            placeholderTextColor={Colors.gray}
          />
        </View>
        <View
          style={{
            flexDirection: "row",

            height: 40,
            backgroundColor: Colors.darkGray,
            borderRadius: 8
          }}
        >
          <KeyIcon style={{ marginLeft: 13, marginTop: 7 }} />
          <TextInput
            onChangeText={value => this.onChangeText("password", value)}
            style={GlobalStyles.destinationInput}
            secureTextEntry
            placeholder="Password"
            placeholderTextColor={Colors.gray}
          />
        </View>
        <View style={{ marginTop: 60, alignItems: "center" }}>
          <Buttons
            title="Sign In"
            backgroundColor={Colors.primary}
            textColor={Colors.black}
            onPress={() => {
              signIn({
                email,
                password
              });
            }}
          />
          <Buttons
            title="Sign Up"
            backgroundColor={Colors.darkGray}
            textColor={Colors.white}
            onPress={() => {
              navigation.navigate("SignUp");
            }}
          />
        </View>
      </View>
    );
  }
}

SignInScreen.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired
  }).isRequired,
  // Redux state
  errorMessage: PropTypes.string,
  loading: PropTypes.bool.isRequired,
  // Redux dispatch
  signIn: PropTypes.func.isRequired
};

SignInScreen.defaultProps = {
  errorMessage: ""
};

const mapStateToProps = state => ({
  errorMessage: state.user.errorMessage,
  loading: state.user.loading
});

const mapDispatchToProps = dispatch => ({
  signIn: credentials => {
    dispatch(userSignIn(credentials));
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SignInScreen);
