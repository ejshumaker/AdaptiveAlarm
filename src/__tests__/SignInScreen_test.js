
/* eslint-disable no-undef */
/* eslint-disable import/first */
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { SignInScreen } from '../screens/SignInScreen';

jest.mock('firebase', () => ({
  auth: () => ({
    onAuthStateChanged: jest.fn(cb => cb({ name: 'Tristan', uid: 1 })),
  }),
}));

Enzyme.configure({ adapter: new Adapter() });


describe('SignIn Screen', () => {
  let wrapper;
  // our mock login function to replace the one provided by mapDispatchToProps
  const mockSignInfn = jest.fn();
  const navigation = { navigate: jest.fn() };

  beforeEach(() => {
    // pass the mock function as the login prop
    wrapper = shallow(<SignInScreen
      navigation={navigation}
      signIn={mockSignInfn}
      loading={false}
      // Redux state
      errorMessage=""
    />);
  });

  it('test SignIn screen matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('test SignIn screen matches snapshot after state set', () => {
    wrapper.setState({
      password: 'testPass',
      email: 'tsteiner4@wisc.edu',
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('should call the sign in action', () => {
    wrapper.setState({
      password: 'testPass',
      email: 'tsteiner4@wisc.edu',
    });
    wrapper.find('[title="Sign In"]').simulate(
      'press',
      { preventDefault() {} },
    );
    expect(mockSignInfn.mock.calls.length).toBe(1);
  });
});
