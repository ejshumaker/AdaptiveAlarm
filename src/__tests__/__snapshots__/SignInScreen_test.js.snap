// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn Screen test SignIn screen matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": undefined,
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "padding": 48,
      },
    ]
  }
>
  <StatusBar
    barStyle="light-content"
  />
  <Text
    style={
      Array [
        Object {
          "color": "#DDDDDD",
          "fontSize": 30,
          "fontWeight": "bold",
        },
        Object {
          "color": "#E4F96A",
          "textAlign": "left",
        },
      ]
    }
  >
    SIGN IN:
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#DDDDDD",
          "fontSize": 18,
          "fontWeight": "300",
        },
        Object {
          "color": "#fa0505",
          "marginVertical": 24,
        },
      ]
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "#505050",
        "borderRadius": 8,
        "flexDirection": "row",
        "height": 40,
        "marginBottom": 30,
      }
    }
  >
    <EmailIcon
      style={
        Object {
          "marginLeft": 13,
          "marginTop": 7,
        }
      }
    />
    <TextInput
      onChangeText={[Function]}
      placeholder="Email"
      placeholderTextColor="#AAAAAA"
      style={
        Object {
          "backgroundColor": "#505050",
          "borderColor": "#505050",
          "borderRadius": 8,
          "color": "#DDDDDD",
          "fontSize": 14,
          "height": 40,
          "marginLeft": 0,
          "marginRight": 5,
          "padding": 10,
          "width": 200,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#505050",
        "borderRadius": 8,
        "flexDirection": "row",
        "height": 40,
      }
    }
  >
    <KeyIcon
      style={
        Object {
          "marginLeft": 13,
          "marginTop": 7,
        }
      }
    />
    <TextInput
      onChangeText={[Function]}
      placeholder="Password"
      placeholderTextColor="#AAAAAA"
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "#505050",
          "borderColor": "#505050",
          "borderRadius": 8,
          "color": "#DDDDDD",
          "fontSize": 14,
          "height": 40,
          "marginLeft": 0,
          "marginRight": 5,
          "padding": 10,
          "width": 200,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": 60,
      }
    }
  >
    <Buttons
      backgroundColor="#E4F96A"
      onPress={[Function]}
      textColor="#13151A"
      title="Sign In"
    />
    <Buttons
      backgroundColor="#505050"
      onPress={[Function]}
      textColor="#DDDDDD"
      title="Sign Up"
    />
  </View>
</View>
`;

exports[`SignIn Screen test SignIn screen matches snapshot after state set 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": undefined,
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "padding": 48,
      },
    ]
  }
>
  <StatusBar
    barStyle="light-content"
  />
  <Text
    style={
      Array [
        Object {
          "color": "#DDDDDD",
          "fontSize": 30,
          "fontWeight": "bold",
        },
        Object {
          "color": "#E4F96A",
          "textAlign": "left",
        },
      ]
    }
  >
    SIGN IN:
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#DDDDDD",
          "fontSize": 18,
          "fontWeight": "300",
        },
        Object {
          "color": "#fa0505",
          "marginVertical": 24,
        },
      ]
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "#505050",
        "borderRadius": 8,
        "flexDirection": "row",
        "height": 40,
        "marginBottom": 30,
      }
    }
  >
    <EmailIcon
      style={
        Object {
          "marginLeft": 13,
          "marginTop": 7,
        }
      }
    />
    <TextInput
      onChangeText={[Function]}
      placeholder="Email"
      placeholderTextColor="#AAAAAA"
      style={
        Object {
          "backgroundColor": "#505050",
          "borderColor": "#505050",
          "borderRadius": 8,
          "color": "#DDDDDD",
          "fontSize": 14,
          "height": 40,
          "marginLeft": 0,
          "marginRight": 5,
          "padding": 10,
          "width": 200,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#505050",
        "borderRadius": 8,
        "flexDirection": "row",
        "height": 40,
      }
    }
  >
    <KeyIcon
      style={
        Object {
          "marginLeft": 13,
          "marginTop": 7,
        }
      }
    />
    <TextInput
      onChangeText={[Function]}
      placeholder="Password"
      placeholderTextColor="#AAAAAA"
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "#505050",
          "borderColor": "#505050",
          "borderRadius": 8,
          "color": "#DDDDDD",
          "fontSize": 14,
          "height": 40,
          "marginLeft": 0,
          "marginRight": 5,
          "padding": 10,
          "width": 200,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": 60,
      }
    }
  >
    <Buttons
      backgroundColor="#E4F96A"
      onPress={[Function]}
      textColor="#13151A"
      title="Sign In"
    />
    <Buttons
      backgroundColor="#505050"
      onPress={[Function]}
      textColor="#DDDDDD"
      title="Sign Up"
    />
  </View>
</View>
`;
